### DISCUSSION TITLE
How to open several URLs with Termux (from a file)?

### DISCUSSION THREAD

**Discussion by ghost** [termux/termux-app#4135] [Q&A] [answered]
Write some URLs to a file:

    $ cat > /data/data/com.termux/files/usr/tmp/example.com

```
https://example.com/
https://example.net/
https://example.org/
https://example.edu/
```

attempt opening them with a script:

```
#!/bin/bash

while IFS= read -r url; do                                   
        termux-open-url "$url"
        sleep 0.5
done < "$1"
```

When running the script, followed by the example text file containing the URLs, only the first URL is reliably opened in the system's default browser.

**Comment by agnostic-apollo** [comment]
Increase sleep time to like `5` seconds and it should probably be fine, this is not a termux issue.

**Reply by ghost** [reply]
Does not function with two or five seconds sleep either, opens the first URL and the script exits.

Sorry if this is not Termux-related, but I do appreciate further help.

**Reply by agnostic-apollo** [reply, marked as answer]
Works fine for me on Android 11. Check if Termux has been granted Draw Over Apps permission as after opening first link, termux would go in background.

**Reply by ghost** [reply]
from Android', 14' Termux 'app info':

![Screenshot_20240903-141421_Settings](https://github.com/user-attachments/assets/2490c909-d18b-42fb-bc58-5ed55a2090e3)
